exonum-java-binding/core/rust/src/proxy/node.rs

```diff
@@ -68,35 +61,29 @@ impl NodeContext {
     }
 
     #[doc(hidden)]
-    pub fn create_snapshot(&self) -> Box<Snapshot> {
-        self.blockchain.snapshot()
+    pub fn create_snapshot(&self) -> Box<dyn Snapshot> {
+        self.api_context.snapshot()
     }
 
     #[doc(hidden)]
     pub fn public_key(&self) -> PublicKey {
-        self.public_key
+        self.api_context.service_keypair().0.clone()
     }
 
     #[doc(hidden)]
-    pub fn submit(&self, transaction: RawTransaction) -> Result<Hash, failure::Error> {
-        let service_id = transaction.service_id();
-
-        if !self.blockchain.service_map().contains_key(&service_id) {
-            return Err(format_err!(
-                "Unable to broadcast transaction: service(ID={}) not found",
-                service_id
-            ));
-        }
-
-        let msg = Message::sign_transaction(
-            transaction.service_transaction(),
-            service_id,
-            self.blockchain.service_keypair.0,
-            &self.blockchain.service_keypair.1,
-        );
-        let tx_hash = msg.hash();
-
-        self.transaction_sender.broadcast_transaction(msg)?;
+    pub fn submit(&self, transaction: Verified<AnyTx>) -> Result<Hash, failure::Error> {
+        // TODO: check service is active
```

Comment:
@alekseysidorov Why doesn't the core check that immediately when a transaction is submitted?

Reply:
@alekseysidorov 🏓

Reply:
Dispatcher finds proper runtime for transaction by their instance ID in lookup table, so if runtime has not specified service instance that means that there is a bug in dispatcher or runtime.